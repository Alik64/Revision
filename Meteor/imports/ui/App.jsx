import React from "react";
import { useTracker } from "meteor/react-meteor-data";
import { Task } from "./Task";
import { TasksCollection } from "../api/TasksCollection.js";
import { TaskForm } from "./TaskForm";

const toggleChecked = ({ _id, isChecked }) => {
  Meteor.call(
    "tasks.updateChecked",
    { _id, isChecked: !isChecked },
    (error) => {
      if (error) {
        alert(`Error updating task: ${error.error}`);
      }
    }
  );
};

const deleteTask = ({ _id }) => {
  Meteor.call("tasks.remove", _id, (error) => {
    if (error) {
      alert(`Error deleting task: ${error.error}`);
    }
  });
};

export const App = () => {
  const { tasks, isLoading, isError } = useTracker(() => {
    const handle = Meteor.subscribe("tasks");
    const tasks = TasksCollection.find({}, { sort: { createdAt: -1 } }).fetch();
    const isLoading = !handle.ready();
    const isError = !tasks;

    return { tasks, isLoading, isError };
  });

  if (isLoading) {
    return <div>Loading...</div>;
  }

  if (isError) {
    return <div>Error loading tasks.</div>;
  }

  console.log(tasks); // Check what tasks are fetched

  return (
    <div>
      <h1>Welcome to Meteor!</h1>
      <TaskForm />
      <ul>
        {tasks.map((task) => (
          <Task
            key={task._id}
            task={task}
            onCheckboxClick={toggleChecked}
            onDeleteClick={deleteTask}
          />
        ))}
      </ul>
    </div>
  );
};
