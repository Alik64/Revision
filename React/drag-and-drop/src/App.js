import React from "react";
import { DragDropContext } from "react-beautiful-dnd";
import initialData from "./initial-data";
import Column from "./Column";

const App = () => {
  const columns = initialData.columnOrder.map((columnId) => {
    const column = initialData.columns[columnId];
    const tasks = column.tasksId.map((taskId) => initialData.tasks[taskId]);
    return <Column key={column.id} column={column} tasks={tasks} />;
  });
  const onDragEndHandler = (result) => {
    // TODO :: Reorder our columns
  };
  return (
    <DragDropContext onDragEnd={onDragEndHandler}>
      <div>{columns}</div>
    </DragDropContext>
  );
};

export default App;
