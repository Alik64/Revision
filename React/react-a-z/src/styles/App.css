.App {
    width: 800px;
}