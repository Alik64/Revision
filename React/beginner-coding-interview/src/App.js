
import { Link, Route, Routes } from 'react-router-dom';
import './App.css';

import About from './ReactRouterDom/pages/About';
import Home from './ReactRouterDom/pages/Home';


function App() {
  return (
    <>
      <Link to='/'>Home</Link>
      <Link to='about'>About</Link>

      <Routes>
        <Route path='/' element={<Home />} />
        <Route path='about' element={<About />} />
      </Routes>
    </>

  );
}

export default App;
