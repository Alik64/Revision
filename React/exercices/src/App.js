import React, { useEffect, useState } from "react";
import axios from "axios";
import "./App.css";
import FocusableInput from "./_TD/FocusableInput";

// const users = [
//   { id: 1, firstname: "Alexandre", lastname: "Zegrand" },
//   { id: 2, firstname: "Marie", lastname: "Rose" },
//   { id: 3, firstname: "India", lastname: "Loulou" },
// ];

const users = [
  { firstName: "Ada", lastName: "Lovelace" },
  { firstName: "Donald", lastName: "Knuth" },
];

function App() {
  const [characters, setCharacters] = useState([]);
  console.table(characters);
  useEffect(() => {
    const fetchData = async () => {
      const result = await axios.get(
        "https://triple-triad-game.herokuapp.com/api/v1/characters"
        // "http://localhost:5001/api/v1/characters"
      );
      console.log(result.data);
      setCharacters(result.data.data);
    };
    fetchData();
  }, []);
  console.log("characters ==>", characters);
  // return (
  //   <>
  //     <div
  //       style={{
  //         display: "grid",
  //         gridTemplateColumns: "repeat(3,1fr)",
  //         gridTemplateRows: "repeat(3,1fr)",
  //         gap: "1rem",
  //       }}
  //     >
  //       {characters.map((char) => (
  //         <div key={char.id}>
  //           <div>
  //             {char.isLike ? "♥" : "♡"}
  //             <h1>{char.name}</h1>
  //             <h3>{char.humanName}</h3>
  //             <img
  //               src={char.thumbnail.path}
  //               alt=""
  //               style={{ width: "150px", height: "150px" }}
  //             />
  //             <p>{char.description}</p>
  //           </div>
  //         </div>
  //       ))}
  //     </div>
  //   </>
  // );
  return <FocusableInput shouldFocus={true} />;
}
export default App;
