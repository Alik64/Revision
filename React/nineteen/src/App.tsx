import { useState } from 'react'

import './App.css'

import Nineteen from './Nineteen';

function App() {
  const [count, setCount] = useState(0)

  return (
   <Nineteen/>
  )
}

export default App
