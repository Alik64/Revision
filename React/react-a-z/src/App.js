import React, { useRef, useState } from "react";
import ClassCounter from "./components/ClassCounter";
import Counter from "./components/Counter";
import PostItem from "./components/PostItem";
import PostList from "./components/PostList";
import MyButton from "./components/UI/button/MyButton";
import MyInput from "./components/UI/input/MyInput";
import './styles/App.css'

function App() {
  const [posts, setPosts] = useState([
    { id: 1, title: 'JavaScript', body: 'Description' },
    { id: 2, title: 'JavaScript', body: 'Description' },
    { id: 3, title: 'JavaScript', body: 'Description' },
    { id: 4, title: 'JavaScript', body: 'Description' },
    { id: 5, title: 'JavaScript', body: 'Description' },
  ])

  const [post, setPost] = useState({ title: '', body: '' })


  const addNewPost = (e) => {
    e.preventDefault()
    const newPost = { ...post, id: Date.now() }
    setPosts([...posts, newPost])



  }
  return (
    <div className="App">

      <form >
        <MyInput
          type="text"
          placeholder="Titre de l'article"
          value={post.title}
          onChange={e => setPost({ ...post, title: e.target.value })}
        />
        <MyInput
          type="text"
          placeholder="Description"
          value={post.body}
          onChange={e => setPost({ ...post, body: e.target.value })}


        />

        <MyButton onClick={addNewPost}>Cr√©er</MyButton>

      </form>
      <PostList posts={posts} title='JavaScript' />


    </div>
  );
}

export default App;
