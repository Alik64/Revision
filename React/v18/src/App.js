import logo from './logo.svg';
import './App.css';
import List from './List';
import { useState, useTransition } from 'react';
import Batching from './Batching';

const source = new Array(12000).fill('').map((_, i) => ({ id: i, name: `Item ${i + 1}` }))

const filterItems = (filter) => {

  return filter
    ? source.filter(i => i.name.includes(filter))
    : source
}


function App() {
  const [filter, setFilter] = useState('')
  const [isPending, startTransition] = useTransition()

  const items = filterItems(filter)

  // const changeHandler = (e) => {
  //   startTransition(() => {
  //     setFilter(e.target.value)
  //   })
  // }
  const changeHandler = (e) => {

    setFilter(e.target.value)

  }
  return (
    <div className="App">
      <input type='text' onChange={changeHandler} />
      <h2>{isPending && 'is Pending...'}</h2>
      <List items={items} />
      <Batching />
    </div>
  );
}

export default App;
