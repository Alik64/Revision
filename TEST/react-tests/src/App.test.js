import { render, screen } from '@testing-library/react';
import App from './App';

describe('<App/>',()=>{
  test('renders my elements', () => {
    render(<App />);
    const myHeading = screen.getByText(/React testing/i);
    const btn = screen.getByRole('button')
    const input = screen.getByPlaceholderText(/hello/i)
    expect(myHeading).toBeInTheDocument()
    expect(input).toBeInTheDocument()
    expect(btn).toBeInTheDocument()
    screen.debug()
    expect(input).toMatchSnapshot()
  });
  test('testing queryByText()', ()=>{
    const hwElem = screen.queryByText(/hello world/i)
    expect(hwElem).toBeNull()
  })

  test('testing async with findBy',()=>{
    const myAsyncElem = screen.findByText(/hello/i)
    
  })
})

 