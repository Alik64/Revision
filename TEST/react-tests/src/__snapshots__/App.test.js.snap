// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> renders my elements 1`] = `
<input
  placeholder="hello"
  type="text"
/>
`;

exports[`renders my elements 1`] = `
<input
  placeholder="hello"
  type="text"
/>
`;
